edgelet/doc/devguide.md

```diff
@@ -1,163 +1,264 @@
 # IoT Edge Security Daemon - Dev Guide
 
-## Building
-The daemon is written in [Rust](https://www.rust-lang.org/en-US/). Rust uses two cli tools for its build: rustup and cargo.
+There are two options for building the IoT Edge Security Daemon.
 
-### Rustup
-Rustup is rust's toolchain manager. Rust ships stable, beta, and nightly versions of the compiler and standard library.
-This project uses stable, but some of the tooling uses the nightly compiler. Rustup is used to update the compiler, and
-switch the active toolchain. You will use rustup infrequently.
+1. Build the OS packages. If you just want to build something that you can install on your device, and are not making frequent changes to the daemon's source code, you probably want this option.
 
-To install rustup, visit [https://www.rustup.rs](https://www.rustup.rs/) and follow the instructions.
+1. Build the daemon binaries. If you want to make frequent changes to the daemon's source code and run it without installing it, you probably want this option.
 
-To install or update the compiler:
+
+## Building OS packages
+
+### Linux
+
+Linux packages are built using the `edgelet/build/linux/package.sh` script. Set the following environment variables, then invoke the script:
+
+1. `PACKAGE_OS`: This is the OS on which the resulting packages will be installed. It should be one of `centos7`, `debian8`, `debian9`, `debian10`, `ubuntu16.04` or `ubuntu18.04`
+
+1. `PACKAGE_ARCH`: This is the architecture of the OS on which the resulting packages will be installed. It should be one of `amd64`, `arm32v7` or `aarch64`.
+
+For example:
+
+```sh
+git clone --recurse-submodules 'https://github.com/Azure/iotedge'
+cd iotedge/
+
+PACKAGE_OS='debian10' PACKAGE_ARCH='arm32v7' ./edgelet/build/linux/package.sh
```

Comment:
We also have a [Makefile](https://github.com/Azure/iotedge/blob/master/edgelet/Makefile) to do things with source code. Should we mention the `make release` command rather than exact script call?

Reply:
The point of this change is the makefile is the wrong thing to use when you want to make packages.

Eg with `make release` on Debian 8 you'll end up using the wrong value for `HOST_URI_OPTION` (the default empty one).

Eg if you go further ahead and use `make install` on any Debian or Ubuntu, you'll end up having a misconfigured iotedged, because despite its generic-sounding name `make install` is only used for CentOS. ( https://github.com/Azure/iotedge/issues/1562 )

The script is there precisely to build the packages properly. That's why the preamble lists two ways of working on iotedged - use the script if you just want packages, or use `cargo` if you want to hack the code.

Reply:
I see. It is confusing. Thanks for explanation.
Wonder if we need to spend some time and fix makefile in a way it does what it is intended to do.

Reply:
I don't think I'd want to bake OS detection in a makefile. The script can be more deterministic because it knows which OS it's spawning a Docker container for.

The script also helps because you don't have to install and configure the build dependencies, esp cross-compiler toolchains. Also, for some of the targets, the build needs to run in a Docker image anyway (the ones that use Linaro images).

Reply:
Makes sense. Agree